File: 3461-find-the-minimum-area-to-cover-all-ones-i/3461-find-the-minimum-area-to-cover-all-ones-i.cpp
class Solution {
public:
    int minimumArea(vector<vector<int>>& grid) {
        pair<int, int> topLeft = {-1, -1};
        pair<int, int> bottomRight = {-1, -1};
        for(int i = 0; i < grid.size(); i++)
        {
            for(int j = 0; j < grid[i].size(); j++)
            {
                if(grid[i][j] == 1)
                {
                    if(topLeft.first == -1)
                    {
                        topLeft = {i, j};
                        bottomRight = {i, j};
                    }
                    else
                    {
                        topLeft.second = min(topLeft.second, j);
                        bottomRight = {i, max(bottomRight.second, j)};
                    }
                }
            }
        }
        return (bottomRight.first - topLeft.first + 1) * (bottomRight.second - topLeft.second + 1);
    }
};

File: 3461-find-the-minimum-area-to-cover-all-ones-i/3461-find-the-minimum-area-to-cover-all-ones-i.py
class Solution:
    def minimumArea(self, grid: List[List[int]]) -> int:
        topLeft = None
        bottomRight = None
        for i in range(len(grid)):
            for j in range(len(grid[i])):
                if grid[i][j] == 1:
                    if not topLeft:
                        topLeft = (i, j)
                        bottomRight = (i, j)
                    else:
                        topLeft = (topLeft[0], min(topLeft[1], j))
                        bottomRight = (i, max(bottomRight[1], j))
        return (bottomRight[0] - topLeft[0] + 1) * (bottomRight[1] - topLeft[1] + 1)
